classes/Tools.php

```diff
@@ -792,10 +792,8 @@ protected static function getContextLocale(Context $context)
             return $locale;
         }
 
-        $container = isset($context->controller) ? $context->controller->getContainer() : null;
-        if (null === $container) {
-            $container = SymfonyContainer::getInstance();
-        }
+        $containerFinder = new ContainerFinder();
+        $container = $containerFinder->getContainer($context);
```

Comment:
You could also attach the container here to `$this->container` if not set already.

Reply:
@eternoendless But there is no property container in the class Tools. May be you want to attach to the property container of the class Context. Isn't it ?

Reply:
Whoops, yes, that's what I meant 😅

Reply:
Done ;)